Sources/ScreenObject/ScreenObject.swift

```diff
@@ -6,46 +6,79 @@ import XCTest
 
+    /// The default time used when waiting.
+    public static let defaultWaitTimeout: TimeInterval = 20
+
     public enum WaitForScreenError: Equatable, Error {
         case timedOut
     }
 
+    /// The possible errors the initialization process can throw.
+    public enum InitError: Equatable, Error {
+        case emptyExpectedElementGettersArray
+    }
+
     /// The `XCUIApplication` instance this screen is part of. This is the value passed at
     /// initialization time.
     public let app: XCUIApplication
 
-    private let expectedElementGetter: (XCUIApplication) -> XCUIElement
+    private let expectedElementGetters: [(XCUIApplication) -> XCUIElement]
 
     /// The `XCUIElement` used to evaluate whether the screen is loaded at runtime.
-    public var expectedElement: XCUIElement { expectedElementGetter(app) }
+    public var expectedElement: XCUIElement {
+        guard let getter = expectedElementGetters.first else {
+            preconditionFailure("`expectedElementGetters` array was empty. This should never occur!")
+        }
+        return getter(app)
+    }
 
     /// Whether the screen is loaded at runtime. Evaluated inspecting the `expectedElement`
     /// property.
     public var isLoaded: Bool { expectedElement.exists }
 
     private let waitTimeout: TimeInterval
 
+    public init(
+        expectedElementGetters: [(XCUIApplication) -> XCUIElement],
+        app: XCUIApplication = XCUIApplication(),
+        waitTimeout: TimeInterval = defaultWaitTimeout
+    ) throws {
+        guard expectedElementGetters.isEmpty == false else {
+            throw InitError.emptyExpectedElementGettersArray
+        }
+
+        self.app = app
+        self.expectedElementGetters = expectedElementGetters
+        self.waitTimeout = waitTimeout
+        try waitForScreen()
+    }
+
+    // Notice that this is a designated initializer, too, so that subclasses can delegate to either
+    // the single- and multiple-getter `init` version. If this had been a `convenience`, subclasses
+    // would not have been able to call it with `super`.
     public init(
         expectedElementGetter: @escaping (XCUIApplication) -> XCUIElement,
         app: XCUIApplication = XCUIApplication(),
-        waitTimeout: TimeInterval = 20
+        waitTimeout: TimeInterval = defaultWaitTimeout
     ) throws {
         self.app = app
-        self.expectedElementGetter = expectedElementGetter
+        self.expectedElementGetters = [expectedElementGetter]
         self.waitTimeout = waitTimeout
         try waitForScreen()
     }
 
     @discardableResult
     func waitForScreen() throws -> Self {
         try XCTContext.runActivity(named: "Confirm screen \(self) is loaded") { (activity) in
-            let result = waitFor(
-                element: expectedElement,
-                predicate: "isEnabled == true",
-                timeout: self.waitTimeout
-            )
+            try expectedElementGetters.forEach { getter in
```

Comment:
This runs the `waitFor...` check on every element in the `expectedElementGetters` array. The `try` at the start is required by the compiler because `waitFor...` can throw an error (`throws`).

Notice that there is no `do { } catch { }` in this function. Because the function is declared with `throws`, if and error is thrown inside it, it will be sent to the function caller. 

"Bubbling" errors is a practical way for low-level or library code like this one to remain flexible and let the consumer decide how best to deal with error.